Pcap++/header/PcapDevice.h

```diff
@@ -18,6 +18,79 @@ namespace pcpp
 	// Forward Declaration - required for IPcapDevice::matchPacketWithFilter
 	class GeneralFilter;
 
+	namespace internal
+	{
+		/// @class PcapHandle
+		/// @brief A wrapper class for pcap_t* which is the libpcap packet capture descriptor.
+		/// This class is used to manage the lifecycle of the pcap_t* object
+		class PcapHandle
+		{
+		public:
+			/// @brief Creates an empty handle.
+			PcapHandle() = default;
+			/// @brief Creates a handle from the provided pcap descriptor.
+			/// @param pcapDescriptor The pcap descriptor to wrap.
+			explicit PcapHandle(pcap_t* pcapDescriptor);
+
+			PcapHandle(const PcapHandle&) = delete;
+			PcapHandle(PcapHandle&& other) noexcept;
+
+			PcapHandle& operator=(const PcapHandle&) = delete;
+			PcapHandle& operator=(PcapHandle&& other) noexcept;
```

Comment:
No one is using it? Can it be deleted?

Reply:
Eh, it can, but I would prefer to keep it for completeness sake. The assignment to nullptr is used, so it would make little sense to have that but not regular assignment.

Reply:
Looking at it, we need to keep the copy assignment as deleted anyway, as it will be implicitly declared otherwise. At that point the only thing we can remove is the move assignment, which I personally would prefer to keep even if its not used at the moment due to rule of 5. This being a lifecycle class, I think it would be beneficial to have all the ctor/assignment semantics be strictly defined from the start.

Reply:
Sorry for confusing you, I am not against the moving assignment declaration. I meant we should mark it as "delete" since it seems to be not used by anyone.

Reply:
I see. Hmm, we have a move constructor tho? Wouldn't having that but not move assignment be confusing? I usually consider those 2 a pair.

Reply:
It seems that the move c'tor is not being used as well... Anyway, I am fine with them. I just though maybe we don't need them.

Reply:
I tried to have the ownership semantics of a unique_ptr, so a move operation would be a transfer of ownership.

Reply:
I see. No problem.